app/controllers/podcast_metrics_controller.rb

```diff
@@ -195,46 +213,31 @@ def set_podcast
     render_not_found(e)
   end
 
-  def set_date_range
-    @date_start = metrics_params[:date_start]
-    @date_end = metrics_params[:date_end]
-    @interval = metrics_params[:interval]
-    @date_range = generate_date_range(@date_start, @date_end, @interval)
+  def metrics_params
+    params
+      .permit(:podcast_id, :episode_id, :prev_episode_id)
   end
 
-  def set_uniques
-    @uniques_selection = uniques_params[:uniques_selection]
-  end
+  def calculate_episode_trend(episode, prev_episode)
+    return nil unless episode.in_default_feed? && episode.first_rss_published_at.present? && prev_episode.present?
+    return nil if (episode.first_rss_published_at + 1.day) > Time.now
 
-  def set_dropday_range
-    @dropday_range = dropdays_params[:dropday_range]
-  end
+    ep_dropday_sum = episode_dropday_query(episode)
+    previous_ep_dropday_sum = episode_dropday_query(prev_episode)
 
-  def metrics_params
-    params
-      .permit(:podcast_id, :date_start, :date_end, :interval)
-      .with_defaults(
-        date_start: 30.days.ago.utc_date,
-        date_end: Date.utc_today,
-        interval: "DAY"
-      )
-  end
+    return nil if ep_dropday_sum <= 0 || previous_ep_dropday_sum <= 0
 
-  def uniques_params
-    params
-      .permit(:uniques_selection)
-      .with_defaults(
-        uniques_selection: "last_7_rolling"
-      )
-      .merge(metrics_params)
+    ((ep_dropday_sum.to_f / previous_ep_dropday_sum.to_f) - 1).round(2)
```

Comment:
Shouldn't this be `round(3)` - since you're displaying 3 digits in the UI?  Seeing everything with a `.0` on it, which seems unlikely in every case.